langchain_postgres/vectorstores.py

```diff
@@ -97,6 +97,124 @@ class DistanceStrategy(str, enum.Enum):
     .union(SPECIAL_CASED_OPERATORS)
 )
 
+class IndexManager:
+    """Manages the creation, listing, and retrieval of indexes for the embedding column in a PostgreSQL database.
+
+    This class provides both synchronous and asynchronous methods to interact with the database, allowing for
+    the creation of different types of indexes (e.g., HNSW, IVFFlat) with various distance functions (e.g., l2, cosine).
+
+    Args:
+        connection (Union[str, Engine, AsyncEngine]): The database connection string or engine instance.
+        async_mode (bool): Flag to indicate if asynchronous operations should be used. Defaults to False.
+    """
+    def __init__(self, connection: Union[str, Engine, AsyncEngine], async_mode: bool = False):
+        self.async_mode = async_mode
+        if isinstance(connection, str):
+            if async_mode:
+                self._engine = create_async_engine(connection)
+            else:
+                self._engine = create_engine(connection)
+        elif isinstance(connection, (Engine, AsyncEngine)):
+            self._engine = connection
+        else:
+            raise ValueError("Invalid connection type")
+
+    def list_indexes(self) -> List[Dict[str, Any]]:
+        """List all indexes."""
+        with self._engine.connect() as conn:
+            result = conn.execute(text("SELECT * FROM pg_indexes WHERE tablename = 'langchain_pg_embedding'"))
```

Comment:
Wouldn't this return all the indexes of the `langchain_pg_embedding` table, not just for the `embedding` column? There are also indexes for the `id` and `cmetadata` columns.

Reply:
You are right I missed to add the argument to select only the `embedding` column. Hopefully this resolves that oversight. Do let me know if I have overlooked something else.